### PR TITLE
Regression intro improvement

### LINKED ISSUE #129: Improve regression introduction
Lead into regression more gentle and rely less on past knowledge about classification (8.3 is a bit bizarre, should combine with 8.4). This will address comments made by Rev C.  **MINOR**

### PR DESCRIPTION
**Pull request by trevorcampbell**
This PR resolves #129. The big change was just to expand the text at the beginning of regression 1 and 2 to make the flow into the content a bit more gentle. Comments / suggestions are welcome.

### CODE REVIEW

**Review by leem44 — COMMENTED** [review]
I only looked at the first few paragraphs of these two chapters (as opposed to going through the entire chapter). Looks really good! I think its a lot clearer now and a much better introduction to regression

File: 07-regression1.Rmd [hunk 1]
```diff
@@ -20,23 +30,35 @@ By the end of the chapter, students will be able to:
...
+we first split the data into training and test sets, in order to ensure that we assess the performance
+of our method on observations not seen during training; and we can use cross-validation to evaluate difference
+choices of model parameters (e.g., K in a K-nearest neighbours model). The major difference
+is that we are now predicting numerical variables instead of categorical variables.
+
+## Exploring a labelled data set
+
+In this chapter and the next, we will study the Sacramento real estate data set.
+This data set was [originally reported in the Sacramento Bee](https://support.spatialkey.com/spatialkey-sample-csv-data/),
 but we have provided it with this repository as a stable source for the data.
```
[the remark this answers is not in the document]
Comment (trevorcampbell) [reply 07-regression1.Rmd:61]: good catch -- I'll remove this comment!

[the remark this answers is not in the document]
Comment (trevorcampbell) [reply 07-regression1.Rmd]: removed

File: 07-regression1.Rmd [hunk 2]
```diff
@@ -1,11 +1,21 @@
...
-This chapter will provide an introduction to regression through K-nearest
-neighbours (K-NN) in a predictive context, focusing primarily on the case where
-there is a single predictor and single response variable of interest. The
-chapter concludes with an example of K-nearest neighbours regression with
-multiple predictors.
+
+This chapter continues our foray into answering predictive questions.
+But unlike the past two chapters, which focused on predicting categorical
+variables, this chapter will focus on predicting numerical
+variables. This process---known as *regression*---has many similarities
```
[the remark this answers is not in the document]
Comment (trevorcampbell) [reply 07-regression1.Rmd:8]: agreed, will make that change now

File: 07-regression1.Rmd [hunk 3]
```diff
@@ -20,23 +30,35 @@ By the end of the chapter, students will be able to:
...
-
-Let's start by loading the packages we need and doing some preliminary
-exploratory analysis. The Sacramento real estate data set we will study in this chapter
-was [originally reported in the Sacramento Bee](https://support.spatialkey.com/spatialkey-sample-csv-data/),
+exercise each week to predict their race time in the annual Boston marathon. As 
+another example, we could try to use the size of a house to
+predict its sale price. Both of these response variables---race time and sale price---are 
+numerical, and so predicting them given past data is considered a regression problem.
+
+> You can tell whether a variable is numerical or categorical---and therefore whether you
```
[the remark this answers is not in the document]
Comment (trevorcampbell) [reply 07-regression1.Rmd:44]: Disagree -- class grades are not categories, they are discretized numerical variables (A = 80 -100, B = 70 - 80, etc). Another way to see this is that a B grade is more similar to A than a C grade is.

File: 07-regression1.Rmd [hunk 4]
```diff
@@ -48,8 +70,14 @@ sacramento <- read_csv("data/sacramento.csv")
...
 ```
 
-The purpose of this exercise is to understand whether we can we use house size
-to predict house sale price in the Sacramento, CA area. The columns in this
+We first need to formulate a precise question that
+we want to answer. In this example, our question is predictive:
+can we use the size of a house in the Sacramento, CA area to predict
+its sale price? A rigorous, quantitative answer to this question might help
+a realtor advise a client as to whether the price of a particular listing 
+is fair, or perhaps how to set the price of a new listing.
```
[the remark this answers is not in the document]
Comment (trevorcampbell) [reply 07-regression1.Rmd:80]: Agree -- I will make a global issue about these!

File: 07-regression1.Rmd [hunk 5]
```diff
@@ -48,8 +70,14 @@ sacramento <- read_csv("data/sacramento.csv")
...
+
```
[the remark this answers is not in the document]
Comment (trevorcampbell) [reply 07-regression1.Rmd:81]: I prefer not to get into the weeds here about this. Instead I'll just choose my words more carefully here.

File: 08-regression2.Rmd [hunk 6]
```diff
@@ -16,27 +19,39 @@ By the end of the chapter, students will be able to:
...
-this chapter we introduce the more general linear regression case where more
-than one predictor can be used.
-
-For example, let's revisit the smaller version of the Sacramento housing data
-set. Recall that we have come across a new 2,000-square foot house we are interested
+the line is used to "look up" the prediction. 
+
+> **Note:** although we did not cover it explicitly in the classification chapters,
+> there is another popular method for classification called *logistic regression* (ignore 
+> the word *regression* there...). After reading this chapter, if you want something like
```
[the remark this answers is not in the document]
Comment (trevorcampbell) [reply 08-regression2.Rmd]: good point. will rephrase!

File: 07-regression1.Rmd (hunk 3 above)
[the remark this answers is not in the document]
Comment (leem44) [reply 07-regression1.Rmd:44]: I agree with you there, but just in terms of using it to help students identify whether to use classification or regression, I wonder if we should rephrase it?

[the remark this answers is not in the document]
Comment (trevorcampbell) [reply 07-regression1.Rmd:44]: Do you have a suggestion? Happy to discuss an alternative!

**Pull request merged**